### ISSUE TITLE
Scribbler: Controlling Deep Image Synthesis With Sketch and Color 

### ISSUE THREAD

**Issue by xchiex17** [DwangoMediaVillage/paper_readings#11]
**CNN＋色ストロークを使ったスケッチからフォトリアリスティックな画像合成**

論文本体・著者
------------------

* Patsorn Sangklo (Georgia Institute of Technology)
* Jingwan Lu (Adobe Research)
* Chen Fang (Adobe Research)
* Fisher Yu (Princeton University)
* James Hays (Georgia Institute of Technology)  

* http://scribbler.eye.gatech.edu/
* CVPR 2017

解きたい問題
------------------
|![image](https://user-images.githubusercontent.com/8101458/28049153-625357dc-6630-11e7-9127-730efb2fdc3e.png)|
|---|
|論文 Eq (2)|
|L<sub>p</sub> : the average per-pixel L2 difference.|
|L<sub>f</sub> : the L2 difference in a feature space, where a feature is extracted from a certain layer of a pre-trained neural network representing highlevel information of images.|
|L<sub>adv</sub> : an adversarial loss.|
|L<sub>tv</sub> : a total variation loss.|

新規性
------------------

* ソースをグレースケール画像に焦点を当てたものが多い中、スケッチに対して画像合成を行った
* 過去手法ではCNNで全自動で色が決めていたため、一部所望の色付けが得られなかったが、任意の物体の色を任意の色に指定できるようにした

実装
------------------
### ネットワーク
![image](https://user-images.githubusercontent.com/8101458/28049339-a7551568-6631-11e7-95d2-e70e3c4846fa.png)
* 3段階のダウンサンプリング
* 7つのレジディアルブロック
* 3つのアップサンプリングからなるエンコードデコーダ
* GAN

### ネットワーク
170万枚枚

### 学習データペア
![image](https://user-images.githubusercontent.com/8101458/28049362-e65d1e72-6631-11e7-9679-5264b3c5b0bb.png)
* スケッチは最初XDoGを使って、写真から生成
* その後にスケッチからの合成にロバスト性を生むために色んな複数のスケッチを混ぜる

### カラーストローク対応用の学習データペア
* カラーストロークに対応するために、学習時にカラーストロークを自動生成して用いた
* カラーストロークは正解のカラー画像から色を取り、疑似的に生成

 
実験・議論
------------------
### 結果
|![image](https://user-images.githubusercontent.com/8101458/28049413-52e06036-6632-11e7-9fdf-3622bd12265d.png)|
|---|
|http://scribbler.eye.gatech.edu/img/results/stroke2photo_gen.png|
![image](https://user-images.githubusercontent.com/8101458/28049458-7c52e84e-6632-11e7-9595-f613b13145f0.png)
* ストロークがないものを見ると彩度の低そうな色を選んでいることがわかる
* ヒントを入れてるものは適切にカラーストロークの色を反映できている

### Limitation & Future work
* 色の滲み出し問題 (the color leaking problem)が発生することがある(Fig.7の車の例など)
* カラーストロークによる、 ground-truthとのlossとadversarial lossの最小化のコントロールが難しい
* データセットの問題で訓練中と同様のスケールのオブジェクトしか実験をしていない

読んだ中での不明点などの感想
------------------

* 評価が特になかった、珍しいと思った

関連論文
------------------
色ストロークを用いた写真の自動彩色
https://github.com/DwangoMediaVillage/paper_readings/issues/8
